src/processors/registration-processor.ts

```diff
@@ -36,22 +38,36 @@ export class RegistrationProcessor implements IRegistrationProcessor {
 
   constructor(
     @Inject('IRegisterDataMapper') private readonly registerDataMapper: IRegisterDataMapper,
-    @Inject('IHackathonDataMapper') private readonly hackathonDataMapper: IDataMapper<Hackathon>,
+    @Inject('IActiveHackathonDataMapper') private readonly hackathonDataMapper: IActiveHackathonDataMapper,
     @Inject('IEmailService') private readonly emailService: IEmailService,
+    @Inject('IMailListService') private readonly mailListService: IMailListService,
+    @Inject('BunyanLogger') private readonly logger: Logger,
   ) {}
 
   public async processRegistration(registration: Registration) {
     const result = await this.registerDataMapper.insert(registration);
     const submission = await this.registerDataMapper.submit(registration);
-    await this.sendRegistrationEmail(registration);
+    try {
+      const { data } = await this.registerDataMapper.getCurrent(registration.uid!);
+      const hackathon = await this.hackathonDataMapper.activeHackathon.toPromise();
+      await this.sendRegistrationEmail(data);
+      await this.subscribeUserToMailingList(hackathon.name, data);
+    } catch (error) {
+      this.logger.error(error);
+    }
     return new ResponseBody(
       'Success',
       200,
-      { result: 'Success', data: { registration, result, submission } },
+      { result: 'Success', data: registration },
     );
   }
 
+  /**
+   * @VisibleForTesting
+   */
   public async sendRegistrationEmail(registration: Registration): Promise<[request.Response, {}]> {
+    const substitutions = new Map<string, string>();
+    substitutions.set('pin', registration.pin!.toString());
```

Comment:
This will break if the template isn't updated

Reply:
Not entirely sure what template this refers to

Reply:
There is an email template which contains a placeholder text. If the template doesn’t have that, this line of code will crash. It should be fixed